src/Core/TypeNameGenerator.cs

```diff
@@ -1,15 +1,57 @@
 ﻿using NJsonSchema;
 using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
 
 namespace Nancy.Metadata.Swagger.Core
 {
+    //https://stackoverflow.com/a/41961738
```

Comment:
Could you move this to a new class called Type Translator?

Reply:
I will look once i get time and get back to you